packages/gestalt/src/BannerOverlay.js

```diff
@@ -0,0 +1,262 @@
+ * ![BannerOverlay light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/BannerOverlay.spec.mjs-snapshots/BannerOverlay-chromium-darwin.png)
+ */
+export default function BannerOverlay({
+  message,
+  title,
+  onDismiss,
+  helperLink,
+  primaryAction,
+  offset = { top: 0, bottom: 0 },
+  thumbnail,
+  zIndex,
+}: Props): ReactNode {
+  const { name: colorSchemeName } = useColorScheme();
+  const isDarkMode = colorSchemeName === 'darkMode';
+
+  const isMobileDevice = useDeviceType() === 'mobile';
+
+  let messageTextElement: Element<'span'> | string;
+
+  if (typeof message === 'string') {
+    messageTextElement = message;
+  }
+  if (typeof title === 'string') {
+    messageTextElement = title;
+  }
+
+  // If `message` is a Text component, we need to override any text colors within to ensure they all match
+  const checkTextNode = () => {
+    const messageIsTextNode =
+      typeof message !== 'string' &&
+      Children.only<Element<typeof Text>>(message).type.displayName === 'Text';
+
+    if (messageIsTextNode) {
+      const textColorOverrideStyles = isDarkMode
+        ? styles.textColorOverrideDark
+        : styles.textColorOverrideLight;
+
+      messageTextElement = <span className={textColorOverrideStyles}>{message}</span>;
+    }
+    return messageIsTextNode;
+  };
+
+  const { accessibilityDismissButtonLabel: accessibilityDismissButtonLabelDefault } =
+    useDefaultLabelContext('BannerOverlay');
+
+  const { lightModeBackground, darkModeBackground, textColor } = DEFAULT_COLORS;
+
+  const dismissButtonComponent = !!onDismiss && (
+    <Flex.Item flex="none" alignSelf={isMobileDevice ? 'end' : 'center'}>
+      <IconButton
+        accessibilityLabel={accessibilityDismissButtonLabelDefault}
+        icon="cancel"
+        iconColor="darkGray"
+        onClick={onDismiss}
+        size="xs"
+      />
+    </Flex.Item>
+  );
+  const CTAComponent = primaryAction && (
+    <Flex.Item flex={isMobileDevice ? 'shrink' : 'none'}>
+      {primaryAction.role === 'link' ? (
+        <PrimaryAction
+          accessibilityLabel={primaryAction.accessibilityLabel}
+          href={primaryAction.href}
+          label={primaryAction.label}
+          onClick={primaryAction.onClick}
+          rel={primaryAction?.rel}
+          role="link"
+          size="sm"
+          target={primaryAction?.target}
+        />
+      ) : (
+        <PrimaryAction
+          accessibilityLabel={primaryAction.accessibilityLabel}
+          label={primaryAction.label}
+          onClick={primaryAction.onClick}
+          role="button"
+          size="sm"
+        />
+      )}
+    </Flex.Item>
+  );
+
+  const isMessageTextNode = checkTextNode();
+  return (
+    <Box
+      color={isDarkMode ? darkModeBackground : lightModeBackground}
+      paddingX={4}
+      paddingY={3}
+      rounding={4}
+      borderStyle="shadow"
+      dangerouslySetInlineStyle={
+        isMobileDevice
```

Comment:
You could add the conditional in the property and compact is some more like:

```
 __style: {
                position: 'fixed',
                bottom: isMobileDevice ? offset.bottom : 'unset';
                top: !isMobileDevice ? offset.top : 'unset'
                left: '50%',
                transform: 'translateX(-50%)',
              },

```

Reply:
Neat! Done :)